arrow-integrations-retrofit-adapter/src/main/kotlin/arrow/integrations/retrofit/adapter/io/TaglessAdapterFactory.kt

```diff
@@ -0,0 +1,38 @@
+package arrow.integrations.retrofit.adapter.io
+
+import arrow.Kind
+import arrow.fx.IO
+import arrow.fx.IOPartialOf
+import arrow.fx.extensions.io.async.async
+import arrow.fx.typeclasses.Async
+import arrow.integrations.retrofit.adapter.parseTypeName
+import retrofit2.CallAdapter
+import retrofit2.Retrofit
+import java.lang.reflect.ParameterizedType
+import java.lang.reflect.Type
+
+class TaglessAdapterFactory<F>(private val async: Async<F>) : CallAdapter.Factory() {
+
+  companion object {
+    fun createIO(): TaglessAdapterFactory<IOPartialOf<Throwable>> = TaglessAdapterFactory(IO.async())
+
+    fun <F> create(async: Async<F>): TaglessAdapterFactory<F> = TaglessAdapterFactory(async)
+  }
+
+  override fun get(returnType: Type, annotations: Array<Annotation>, retrofit: Retrofit): CallAdapter<*, *>? {
+    val rawType = getRawType(returnType)
+
+    if (returnType !is ParameterizedType) {
+      val name = parseTypeName(returnType)
+      throw IllegalArgumentException("Return type must be parameterized as " +
+        "$name<Foo> or $name<out Foo>")
+    }
+
+    return if (Kind::class.java.isAssignableFrom(rawType)) {
```

Comment:
You're comparing with Kind, but you only support `F`. Shouldn't this be `rawType == F`? (I know this doesn't compile hahaha). 

🤔 What happens if I request IO and I have added only the Factory for ObservableK? 

If I add multiple factories like this one... one for ObservableK and one for IO, both will pass this check... Won't I have an inconsistent state?

Reply:
I think that solves your problem =] 

```
//Factory method
inline fun <reified F> create(async: Async<F>): TaglessAdapterFactory<F> = TaglessAdapterFactory(async, F::class.java)

//Constructor
class TaglessAdapterFactory<F>(private val async: Async<F>, private val clazz: Class<F>) : CallAdapter.Factory() {

//Comparation
return if (clazz::class.java == rawType) {
```

Reply:
I will try this, thanks